dev/build_arrow.sh

```diff
@@ -70,7 +70,8 @@ function build_arrow_java() {
 
     mvn clean install -pl bom,maven/module-info-compiler-maven-plugin,vector -am \
           -DskipTests -Drat.skip -Dmaven.gitcommitid.skip -Dcheckstyle.skip -Dassembly.skipAssembly
-
+    
+    export CMAKE_BUILD_PARALLEL_LEVEL=112
```

Comment:
Can we use `nproc` or some command to obtain a proper parallelism number? `112` is a magic number.